app/index.js

```diff
@@ -81,6 +81,7 @@ module.exports = class GeneratorLicense extends Generator {
         name: 'license',
         message: 'Which license do you want to use?',
         default: this.options.defaultLicense,
+        when: this.options.license == null || licenses.find(x => x.value === this.options.license) === undefined,
```

Comment:
Well this is analogue to the implementation on the other prompts. If you want this changed, it should be a different PR just touching this part and aligning all the when conditions on the prompts at once.